### DISCUSSION TITLE
Validating if a user is actually logged in (with mastodon and token)

### DISCUSSION THREAD

**Discussion by GerritKuilder** [neet/masto.js#855] [Q&A]
Hello All,

struggling with the following:

I  am trying to validate i a login was succesful or not:
```
  const masto = await login({
      url: url,
      accessToken: token
    });
```

but I cannot figure out how I can check if  the login was successful or not.
The masto object does not give me many clues either.

I am trying to understand the docs but so far no luck.

Any guidance would be appreciated.

Kind Regards,

Gerrit Kuilder

**Comment by neet** [comment]
You can use `masto.v1.accounts.verifyCredentials`

> Test to make sure that the user token works.
> https://docs.joinmastodon.org/methods/accounts/#verify_credentials

Example:

```ts
const masto = await login(...);
const me = await masto.v1.accounts.verifyCredentials();
console.info(`Logged in as ${me.acct}`);
```

I hope it helps.
